src/proc.cpp

```diff
@@ -363,96 +363,80 @@ typedef unsigned int process_generation_count_t;
 /// the SIGCHLD signal handler, and therefore does not need atomics or locks.
 static volatile process_generation_count_t s_sigchld_generation_cnt = 0;
```

Comment:
If we're not using it anymore, why is it still there?

Reply:
We're using it now.

Reply:
Ah, sorry, I was going commit-for-commit, and the message for a5a23c57d82faa60a00dd6f7cd618a80c92c7c7d ("Overhaul behavior of `process_mark_finished_children()`") says:

>We're not using the SIGCHLD count in s_sigchld_generation_cnt for
anything any more, as it's not actually a reliable metric

Reply:
Understood. I actually debated deleting it in that same commit but left it around to see if we would need it later (which proved to be the case). I don't think I want to merge a5a23c5 with the waitpid optimizations commit, I would rather they be two separate, testable/bisectable chunks.

Reply:
Okay.